src/main/java/teammates/ui/template/CourseEditSectionRow.java

```diff
@@ -0,0 +1,225 @@
+    private List<ElementTag> permissionInputGroup3;
+    private ElementTag toggleSessionLevelInSectionButton;
+    private List<CourseEditFeedbackSessionRow> feedbackSessions;
+    private List<List<ElementTag>> specialSections;
+    
+    public CourseEditSectionRow(String sectionName, List<String> sectionNames, int sectionIndex, 
+                                InstructorAttributes instructor, int instructorIndex, 
+                                List<String> feedbackNames) {
+        this.sectionName = sectionName;
+        this.instructor = instructor;
+        feedbackSessions = new ArrayList<CourseEditFeedbackSessionRow>();
+        
+        specialSections = createSpecialSectionsForSectionRow(sectionNames, sectionIndex);
+        permissionInputGroup2 = createPermissionInputGroup2ForSectionRow(sectionIndex);
+        permissionInputGroup3 = createPermissionInputGroup3ForSectionRow(sectionIndex);
+        
+        String content = "";
+        String onClick = "";
+        if (!isSessionsInSectionSpecial()) {
+            content = "Give different permissions for sessions in this section";
+            onClick = "showTuneSessionnPermissionsDiv(" + instructorIndex + ", " + sectionIndex + ")";
+        } else {
+            content = "Hide session-level permissions";
+            onClick = "hideTuneSessionnPermissionsDiv(" + instructorIndex + ", " + sectionIndex + ")";
+        }
+        
+        String id = "toggleSessionLevelInSection" + sectionIndex + "ForInstructor" + instructorIndex;
+        toggleSessionLevelInSectionButton = createButton(content, "small col-sm-5", id, "javascript:;", null,
+                                                         onClick, false);
+        
+        for (String feedbackName : feedbackNames) {
+            CourseEditFeedbackSessionRow feedbackSessionRow = new CourseEditFeedbackSessionRow(feedbackName);
+            
+            String name = Const.ParamsNames.INSTRUCTOR_PERMISSION_SUBMIT_SESSION_IN_SECTIONS 
+                          + Const.ParamsNames.INSTRUCTOR_SECTION_GROUP + sectionIndex + "feedback" 
+                          + feedbackName;
+            boolean isChecked = (instructor != null) 
+                                && instructor.isAllowedForPrivilege(sectionName, feedbackName, 
+                                                Const.ParamsNames.INSTRUCTOR_PERMISSION_SUBMIT_SESSION_IN_SECTIONS);
+            ElementTag submitSessionInSectionCheckBox = createCheckBox(null, name, "true", isChecked);
+            
+            name = Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_SESSION_IN_SECTIONS 
+                   + Const.ParamsNames.INSTRUCTOR_SECTION_GROUP + sectionIndex + "feedback" + feedbackName;
+            isChecked = (instructor != null) 
+                        && instructor.isAllowedForPrivilege(sectionName, feedbackName, 
+                                        Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_SESSION_IN_SECTIONS);
```

Comment:
Not sure if it's a good idea, but would it be tidier to have a `Map<String, Boolean>` with all the privileges set beforehand?
You could just list all the constants in an array or something, then loop trough them to populate the map, instead of having 10 of these
` (instructor != null) && instructor.isAllowedForPrivilege(sectionName, feedbackName, Const.ParamsNames.*);`